tests/testthat/test_imputation/test_imputation.R

```diff
@@ -428,4 +428,38 @@ describe('imputation', {
 
   })
 
+  it('extends time scope beyond individual questions', {
+    # On March 15, we found a bug where the function is (still) applying the
+    # time scope variable only to particular questions. So if John is the
+    # only student in the PERTS Network who takes the survey in the week of
+    # March 7 (week_start == "2021-03-07"), and John only does half of the
+    # survey questions, this triggers imputation for all participants for
+    # the questions he completed, but not for the questions he didn't complete.
+    # So there would be no data for the week of 3/7 for those participants.
+
+    rd <- data.frame(
+      week_start = as.Date(c("2021-02-28", "2021-03-07")),
+      participant_id = c("Participant_1", "Participant_1"),
+      q1 = c(2, 3),
+      q2 = c(4, NA),
+      parent_id = "Network_1"
+    )
+
+    rdi <- imputation$impute_to_time_ordinal(
+      response_data = rd,
+      imputation_index = c("participant_id", "parent_id"),
+      time_ordinal_column = "week_start",
+      cols_to_impute = c("q1", "q2"),
+      time_ordinal_scope_vars = "parent_id"
+    )
+
+    # assert that the imputed value for Participant_1 for q2 for 2021-03-07
+    # should be 4
+    expected_q2_w2 <- rdi %>%
+      filter(week_start %in% as.Date("2021-03-07"),
+             question_code %in% "q2") %>%
+      pull(value_imputed)
+    expect_equal(expected_q2_w2, 4)
+  })
```

Comment:
nit: my weaker understanding of this makes me worried about the participant-crossing case (since this test only has "Participant_1"). Maybe I'll try to just write that test quickly and add it.

Reply:
Ok, I think I wrote the test I wanted. It fails before your fix and passes afterward, no other code changes.

Reply:
sweet! Thanks Chris!